src/domain/task-system/src/entities/task_status.rs

```diff
@@ -33,4 +35,18 @@ pub enum TaskOutcome {
     // Replaced(TaskID),
 }
 
+impl TaskOutcome {
+    pub fn is_success(&self) -> bool {
+        matches!(self, Self::Success(_))
+    }
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum TaskResult {
+    Empty,
+    PullResult(Box<PullResult>),
```

Comment:
Is this boxing forced by clippy or is your choice?

I think in Rust the idiomatic way is to hide memory management inside the objects.

In other words, if `PullResult` becomes too big to keep on the stack (which I don't think it is yet) - then we would box the internal state of the `PullResult`, while allowing all upper layers to use it as a simple movable type.

This way `PullResult` encapsulates how exactly it manages memory, while users of this type only have to respect whether it's `Copy` / `Clone` / `Send`. And when it becomes to big to keep on the stack - it can be constructed directly on the heap, without touching any code that uses it.

I wouldn't want to see our enums sprawling with `Box<>`  and `Arc<>` for different variants because it's really hard to reason about them on this level.